### PR TITLE
Add a gNOI operation to restart an OS process.

### PR DESCRIPTION
**Pull request by greg-dennis**

### CODE REVIEW

File: system/system.proto [hunk 1]
```diff
@@ -311,3 +314,20 @@ message SetPackageRequest {
 
 message SetPackageResponse {
 }
+
+// RestartProcessRequest describes the process restart operation.
+message RestartProcessRequest {
+  // Name of the process to be restarted.
+  string name = 1;
+  // Mode of restarting.
+  enum Mode {
```
Comment (marcushines) [inline system/system.proto:323]: Should these map to os signals?
https://man7.org/linux/man-pages/man7/signal.7.html
or at least use the same names or ?

**Review by robshakir — DISMISSED** [review]

File: system/system.proto (hunk 1 above)
Comment (marcushines) [reply system/system.proto]: lgtm

File: system/system.proto [hunk 2]
```diff
@@ -315,19 +315,26 @@ message SetPackageRequest {
...
+// KillProcessRequest describes the process kill operation.
+// Either a pid or name of the procss must be specified.
+message KillProcessRequest {
+  // Process ID of the process to be killed.
+  uint32 pid = 1;
+  // Name of the process to be killed.
+  string name = 2;
+  // Termination signal sent to the process.
+  enum Signal {
+    SIGNAL_UNSPECIFIED = 0;  // Invalid default.
```
[the remark this answers is not in the document]
Comment (marcushines) [reply system/system.proto:327]: i think we should always specific - it makes it so there is no "user defined default behavior"

**Review by marcushines — APPROVED** [review]

File: system/system.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (greg-dennis) [reply system/system.proto:327]: Clarified the contract so a termination signal is always required.

**Pull request merged**